File: web/src/api/shortLink.js
import service from '@/utils/request'

// @Router /short-link/create [post]
export const shortLinkCreate = (data) => {
    return service({
        url: '/short-link/create',
        method: 'post',
        data
    })
}

// @Router /short-link/list [post]
export const shortLinkList = (data) => {
    return service({
        url: '/short-link/list',
        method: 'post',
        data
    })
}

// @Router /short-link/detail [post]
export const shortLinkDetail = (data) => {
    return service({
        url: '/short-link/detail',
        method: 'post',
        data
    })
}

// @Router /short-link/update [post]
export const shortLinkUpdate = (data) => {
    return service({
        url: '/short-link/update',
        method: 'post',
        data
    })
}

// @Router /short-link/delete [post]
export const shortLinkDelete = (data) => {
    return service({
        url: '/short-link/delete',
        method: 'post',
        data
    })
}

// @Router /short-link/click-log/list [post]
export const clickLogList = (data) => {
    return service({
        url: '/short-link/click-log/list',
        method: 'post',
        data
    })
}

File: server/router/short_link/short_link.go
package short_link

import (
	"context"
	"sync"

	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/flipped-aurora/gin-vue-admin/server/router/short_link/initialize"
	"github.com/gin-gonic/gin"
)

type ShortLinkRouter struct{}

var once sync.Once

func (r *ShortLinkRouter) InitApiRouter(Router *gin.RouterGroup) {
	once.Do(func() {
		ctx := context.Background()
		initialize.Menu(ctx)
		initialize.Api(ctx)
	})

	apiRouter := Router.Group("short-link")
	apiRouter.Use(middleware.OperationRecord())
	{
		apiRouter.POST("create", shortLinkApi.Create)
		apiRouter.POST("list", shortLinkApi.List)
		apiRouter.POST("detail", shortLinkApi.Detail)
		apiRouter.POST("update", shortLinkApi.Update)
		apiRouter.POST("delete", shortLinkApi.Delete)
		apiRouter.POST("click-log/list", shortLinkApi.ClickLogList)
	}
}
